flavors/lineageos/update.sh

```diff
@@ -4,9 +4,29 @@
 
 set -euo pipefail
 
-cd "$(dirname "${BASH_SOURCE[0]}")"
+SKIP_REPO_FILE=""
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --skip-mk-repo-file|-s)
+            SKIP_REPO_FILE=1
+            ;;
+        --)
+            shift
+            branch="$1"
+            break
+            ;;
+        -*)
+            echo "Unknown option: $1"
+            exit 1
+            ;;
+        *)
+            branch="$1"
+            ;;
+    esac
+    shift
+done
 
-branch=$1
+cd "$(dirname "${BASH_SOURCE[0]}")"
```

Comment:
Could you use pushd/popd instead? Also please separate this change into a separate commit.

Otherwise diff LGTM and it works fine in my testing.

Reply:
where would I put the popd though? I could put it at the end of the script, but it doesn't really do anything then, right? (note: this `cd` is not changed by my patch, just moved)

Reply:
Indeed